crates/db_schema/replaceable_schema/triggers.sql

```diff
@@ -615,3 +615,98 @@ CREATE TRIGGER change_values
     FOR EACH ROW
     EXECUTE FUNCTION r.private_message_change_values ();
 
+-- When creating or resolving a report, update a report count
+-- on the post and comment aggregate tables
+CREATE FUNCTION r.update_post_aggregates_report_count ()
+    RETURNS TRIGGER
+    LANGUAGE plpgsql
+    AS $$
+BEGIN
+    UPDATE
+        post_aggregates
+    SET
+        report_count = report_count + 1,
+        unresolved_report_count = unresolved_report_count + 1
+    WHERE
+        post_id = NEW.post_id;
+    RETURN NULL;
+END
+$$;
+
+CREATE TRIGGER post_report_aggregates
+    AFTER INSERT ON post_report
+    FOR EACH ROW
+    EXECUTE FUNCTION r.update_post_aggregates_report_count ();
+
+-- When resolving / unresolving a report, update the unresolved_report_count
+CREATE FUNCTION r.update_post_aggregates_unresolved_report_count ()
+    RETURNS TRIGGER
+    LANGUAGE plpgsql
+    AS $$
+BEGIN
+    UPDATE
+        post_aggregates
+    SET
+        unresolved_report_count = unresolved_report_count + CASE WHEN NEW.resolved THEN
+            -1
+        ELSE
+            1
+        END
+    WHERE
+        post_id = NEW.post_id;
+    RETURN NULL;
+END
+$$;
+
+CREATE TRIGGER post_report_unresolved_aggregates
+    AFTER UPDATE OF resolved ON post_report
+    FOR EACH ROW
+    EXECUTE FUNCTION r.update_post_aggregates_unresolved_report_count ();
```

Comment:
Do it like this instead, which is more consistent with other triggers and correctly handles edge cases (such as updates that include the checked columns but set it to the same value as before):

```
CALL r.create_triggers ('post_report', $$
BEGIN
    UPDATE
        post_aggregates AS a
    SET
        report_count = a.report_count + diff.report_count,
        unresolved_report_count = a.unresolved_report_count + diff.unresolved_report_count
    FROM (
        SELECT
            coalesce(sum(count_diff), 0) AS report_count,
            coalesce(sum(count_diff) FILTER (WHERE NOT (post_report).resolved), 0) AS unresolved_report_count
        FROM select_old_and_new_rows AS old_and_new_rows) AS diff
WHERE
    (diff.report_count, diff.unresolved_report_count) != (0, 0);
RETURN NULL;
END;
$$);
```

Same for comment reports.

Also the new triggers should be with the other count triggers, just above the line that says `-- These triggers create and update rows in each aggregates table to match its associated table's rows.`

Reply:
I tried pasting that, but it doesn't pass the tests, as its not grouping by the post id.

Also it's probably not as efficient, because I think its triggering on any change of post_report, rather than only insert and update on a specific column.

Reply:
This might work:

```
CALL r.create_triggers ('post_report', $$
BEGIN
    UPDATE
        post_aggregates AS a
    SET
        report_count = a.report_count + diff.report_count,
        unresolved_report_count = a.unresolved_report_count + diff.unresolved_report_count
    FROM (
        SELECT
            (post_report).post_id,
            coalesce(sum(count_diff), 0) AS report_count,
            coalesce(sum(count_diff) FILTER (WHERE NOT (post_report).resolved), 0) AS unresolved_report_count
        FROM select_old_and_new_rows AS old_and_new_rows GROUP BY (post_report).post_id) AS diff
WHERE
    (diff.report_count, diff.unresolved_report_count) != (0, 0) AND a.post_id = diff.post_id;
RETURN NULL;
END;
$$);
```

Efficiency for updates that don't involve the resolved column (these updates probably don't happen a lot anyway) should be nearly the same because of the `(diff.report_count, diff.unresolved_report_count) != (0, 0)` condition, which guarantees that there's no additional writing or locking, and almost certainly no additional reading. So it should just be evaluating simple expressions on the old and new row in memory. And using the create_triggers function makes it much harder to make a trigger with subtle bugs like the per-row triggers here.

Reply:
Mmk, yeah its probably not a big performance issue, although it is recommended to do column specific triggers if possible. This did work, your group by there was the key. Updated the PR.